src/main/java/com/epam/Consumer.java

```diff
@@ -0,0 +1,27 @@
+package com.epam;
+
+import org.apache.log4j.Logger;
+
+/**
+ * Created by Almas_Doskozhin
+ * on 2/22/2016.
+ */
+public class Consumer implements Runnable {
```

Comment:
Consumer and producer are quite similiat.
Could we avoid that somehow?
May be strategy pattern?